src/xasl/xasl_predicate.hpp

```diff
@@ -152,13 +153,16 @@ namespace cubxasl
 
   struct pred_expr
   {
-    union
-    {
-      pred m_pred;
-      eval_term m_eval_term;
-      pred_expr *m_not_term;
-    } pe;
-    TYPE_PRED_EXPR type;
+    public:
```

Comment:
Do you intend to transform to class ?
public doesn't seem to be required.

Reply:
fixed